OGC-Core/OGCGeometry.class.st

```diff
@@ -1,43 +1,227 @@
+closed, the resulting boundary can be represented using representational Geometry primitives (Reference [1],
+section 3.12.2). The return type is integer, but is interpreted as Boolean, TRUE=1, FALSE=0."
+	self subclassResponsibility 
+]
+
+{ #category : #'methods analysis' }
+OGCGeometry >> buffer: aDistance [
+	"Returns a geometric object that represents all Points whose distance
+from this geometric object is less than or equal to distance."
+	self subclassResponsibility 
+]
+
+{ #category : #testing }
+OGCGeometry >> contains: anotherGeometry [
+	self subclassResponsibility 
+]
+
+{ #category : #'methods analysis' }
+OGCGeometry >> convexHull [
+	"Returns a geometric object that represents the convex hull of this geometric
+object. Convex hulls, being dependent on straight lines, can be accurately represented in linear interpolations
+for any geometry restricted to linear interpolations."
+	self subclassResponsibility 
+	
+]
+
+{ #category : #testing }
+OGCGeometry >> crosses: anotherGeometry [
+	self subclassResponsibility 
+]
+
+{ #category : #'methods analysis' }
+OGCGeometry >> difference: anotherGeometry [
+	"Returns a geometric object that represents the Point
+set difference of this geometric object with anotherGeometry."
+	self subclassResponsibility 
+]
+
+{ #category : #basic }
+OGCGeometry >> dimension [ 
+	"The inherent dimension of this geometric object, which must be less than or equal to the coordinate dimension."
+	self subclassResponsibility 
+]
+
+{ #category : #testing }
+OGCGeometry >> disjoint: anotherGeometry [
+	self subclassResponsibility 
+]
+
+{ #category : #'methods analysis' }
+OGCGeometry >> distance: anotherGeometry [
+	"Returns the shortest distance between any two Points in
+the two geometric objects as calculated in the spatial reference system of this geometric object. Because the
+geometries are closed, it is possible to find a point on each geometric object involved, such that the distance
+between these 2 points is the returned distance between their geometric objects."
+	self subclassResponsibility 
+]
+
+{ #category : #basic }
+OGCGeometry >> envelope [
+	"The minimum bounding box for this Geometry, returned as a Geometry. The
+polygon is defined by the corner points of the bounding box [(MINX, MINY), (MAXX, MINY), (MAXX, MAXY),
+(MINX, MAXY), (MINX, MINY)]. Minimums for Z and M may be added. The simplest representation of an
+Envelope is as two direct positions, one containing all the minimums, and another all the maximums. In some
+cases, this coordinate will be outside the range of validity for the Spatial Reference System"
+	self subclassResponsibility 
+]
+
+{ #category : #basic }
+OGCGeometry >> geometryType [
+	"Returns the name of the instantiable subtype of Geometry of which this geometric object is an instantiable member"
+	^ self class geometryType
+]
+
+{ #category : #'private-initialization' }
+OGCGeometry >> initialize [
+	spatialRS := OGCGeometricProjection

```

Comment:
Which is the more accurate term? reference system or projection. As you assigned a projection to a referenceSystem it tells something about its meaning